e2e/logs_catcher/matcher.go

```diff
@@ -2,118 +2,208 @@ package logs_catcher
 
 import (
 	"context"
-	"fmt"
-	"time"
-
+	"encoding/json"
 	"errors"
+	"fmt"
 	"go.uber.org/zap"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/bloxapp/ssv/e2e/logs_catcher/docker"
 	"github.com/bloxapp/ssv/e2e/logs_catcher/logs"
 )
 
-// Test conditions:
+const beaconProxyContainer = "beacon_proxy"
 
-const waitTarget = "beacon_proxy"
-const firstTarget = "beacon_proxy"
+var ssvNodesContainers = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
 
-var secondTargets = []string{"ssv-node-1", "ssv-node-2", "ssv-node-3", "ssv-node-4"}
+type Matcher struct {
+	logger *zap.Logger
+	cli    DockerCLI
+	mode   SubMode
+}
+
+func NewLogMatcher(logger *zap.Logger, cli DockerCLI, mode SubMode) *Matcher {
+	return &Matcher{mode: mode, cli: cli, logger: logger}
+}
 
-const waitFor = "End epoch finished"
+// Match starts log catching based on the Matcher mode
+func (m *Matcher) Match(pctx context.Context) error {
+	ctx, cancel := context.WithTimeout(pctx, 24*time.Minute) // 24 minutes for 4 epochs
```

Comment:
Maybe we'll need to run this with custom eth network, so let's calculate this instead